File: C#/1557. Minimum Number of Vertices to Reach All Nodes.cs
public class Solution {
    public IList<int> FindSmallestSetOfVertices(int n, IList<IList<int>> edges) {
        List<int> res = new();
        HashSet<int> endNodes = new();
        
        foreach (var edge in edges)
        {
            endNodes.Add(edge[1]);
        }
        
        for (int i = 0; i < n; i++) {
            if (endNodes.Contains(i) is false)
                res.Add(i);
        }
        return res;
    }
}

File: 1557. Minimum Number of Vertices to Reach All Nodes.cc
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> res;
        vector<int> st(n);
        for (const auto& e : edges) {
            st[e[1]]++;
        }
        for (int i = 0; i < n; i++) {
            if (st[i] == 0)
                res.push_back(i);
        }
        return res;
    }
};